internal/datanode/data_sync_service.go

```diff
@@ -102,6 +104,10 @@ func (dsService *dataSyncService) GracefullyClose() {
 		log.Info("dataSyncService gracefully closing flowgraph")
 		dsService.fg.SetCloseMethod(flowgraph.CloseGracefully)
 		dsService.close()
+
+		// clean up metrics
+		pChan := funcutil.ToPhysicalChannel(dsService.vchannelName)
+		metrics.CleanupDataNodeCollectionMetrics(paramtable.GetNodeID(), dsService.collectionID, pChan)
```

Comment:
Should be cleared at dsService.close(), GracefullyClose() will be called only when dn is closing.

Reply:
done